src/pal/src/misc/time.cpp

```diff
@@ -179,6 +223,17 @@ QueryPerformanceCounter(
         lpPerformanceCount->QuadPart = 
             (LONGLONG)ts.tv_sec * (LONGLONG)tccSecondsToNanoSeconds + (LONGLONG)ts.tv_nsec;
     }
+#elif HAVE_MACH_ABSOLUTE_TIME
+    {
+        // use denom == 0 to indicate that s_TimebaseInfo is uninitialised.
+        if (s_TimebaseInfo.denom == 0)
+        {
+            ASSERT("s_TimebaseInfo is uninitialized.\n");
+            goto EXIT;
+        }
+        lpPerformanceCount->QuadPart = 
+            (LONGLONG)mach_absolute_time() * (LONGLONG)s_TimebaseInfo.numer / (LONGLONG)s_TimebaseInfo.denom;
```

Comment:
You should return mach_absolute_time() directly here, and instead use s_TimebaseInfo to compute return value of QueryPerformanceFrequency.